File: 2525-count-number-of-distinct-integers-after-reverse-operations/count-number-of-distinct-integers-after-reverse-operations.java
import java.util.HashMap;

class Solution {
    public int countDistinctIntegers(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        // Insert original integers and their reversed counterparts into the map
        for (int num : nums) {
            map.put(num, 1);
            int reversed = 0;
            int temp = num;
            while (temp > 0) {
                int rem = temp % 10;
                reversed = reversed * 10 + rem;
                temp /= 10;
            }
            map.put(reversed, 1);
        }
        
        return map.size(); // Return the size of the map, which contains distinct integers
    }
}

File: 2525-count-number-of-distinct-integers-after-reverse-operations/count-number-of-distinct-integers-after-reverse-operations.cpp

class Solution {
    int rev(int n) {
        int rev = 0;
        int temp = n;
        while (temp > 0) {
            int rem = temp % 10;
            rev = rev * 10 + rem;
            temp /= 10;
        }
        return rev;
    }
public:
    int countDistinctIntegers(vector<int>& nums) {
        vector<int> v;
        for (auto it : nums) {
            v.push_back(it);
            int a = rev(it);
            v.push_back(a);
        }
        
        map<int, int> mp;
        for (auto it : v) {
            mp[it]++;
        }
        
        return mp.size(); // Return the size of the map, which represents the count of distinct integers
    }
};
